rules/engine.go

```diff
@@ -69,7 +70,7 @@ type V3Engine interface {
 	AddRule(rule DynamicRule,
 		lockPattern string,
 		callback V3RuleTaskCallback,
-		options ...RuleOption)
+		options ...RuleOption) error
```

Comment:
Humm, this would be a major hit to all services to handle this new error.

Maybe we should consider a `e.logger.Fatal(...)` instead?  That way only those that have issues will affected.